pkg/serverstate/instance.go

```diff
@@ -51,3 +52,14 @@ type InstanceExec struct {
 	// around or not.
 	Context context.Context
 }
+
+type InstanceExecHandler interface {
```

Comment:
Could probably use a header comment here.

Reply:
Oh good call. I ran into a few nits when using that I'll push up shortly too.